base58.c

```diff
@@ -37,39 +49,69 @@ typedef uint32_t b58_almostmaxint_t;
 #define b58_almostmaxint_bits (sizeof(b58_almostmaxint_t) * 8)
 static const b58_almostmaxint_t b58_almostmaxint_mask = ((((b58_maxint_t)1) << b58_almostmaxint_bits) - 1);
 
+//	MSVC 2017 doesn't support the GCC (and probably clang) extension
+//	for dynamic arrays in C
```

Comment:
>Variable length arrays are a standard C99 feature, not an extension. 

That may be, but MSVC still doesn't support them. _alloca or the MSVC _malloca  can be used if that's preferred - these perform stack allocation (though _malloca will allocate for sufficiently large sizes).

Reply:
AFAIK, MSVC is a C++ compiler, not a C compiler.

I suppose it can't hurt to use _malloca or such, based on a failed configure check for standards compatibility...

Reply:
A few years back they added C99 support. It's still subpar in comparison to MinGW/GCC, but it's better than it used to be - enough to where people who like C end up using it over C++.

Reply:
Wrote a simple alternative in #9; can you check if it works?